locator/tablets.hh

```diff
@@ -157,6 +157,7 @@ enum class tablet_transition_stage {
     write_both_read_new,
     use_new,
     cleanup,
+    revert_migration,
     end_migration,
```

Comment:
Since the state machine is now non-linear, please follow up with a mermaid diagram in the appropriate place (I don't think there is one).

Reply:
There's no one :( I'll add one as a followup, because even with this pr the diagram would still be incomplete